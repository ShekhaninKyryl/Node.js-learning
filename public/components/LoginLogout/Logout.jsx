import React, {PureComponent} from 'react';
import {connect} from "react-redux";
import {getLogout} from "../../reducers/Actions/loginTracks";


class Logout extends PureComponent {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <input className='my-button logout-button' type="button" value="Logout" id="login" onClick={this.props.logout}/>
        )
    }
}


export default connect(
    state => ({
        api: state.api
    }),
    dispatch => ({
        logout: () => dispatch(getLogout()),
    })
)
(Logout);
