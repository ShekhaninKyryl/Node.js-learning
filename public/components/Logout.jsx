import React, {Component} from 'react';
import axios from 'axios';
import ReactDOM from "react-dom";
import {BrowserRouter as Router, Route, Link} from "react-router-dom";
import Input from "./Input.jsx";


class Logout extends Component {
  constructor(props) {
    super(props);
    this.Logout = this.Logout.bind(this);
  }

  //todo axios
  Logout() {
    let {Logout} = this.props;

    axios.get('/api/logout')
      .then(response => Logout(response.data))
      .catch(error => console.log('Logout error:', error.response));

  }


  render() {

    return (
      <input style={
        {
          position: 'absolute',
          right: 0,
          top: 0
        }
      }
             type="button" value="Logout" id="login" onClick={this.Logout}/>
    )
  }
}


export default Logout;
