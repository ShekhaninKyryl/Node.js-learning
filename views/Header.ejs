<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
<title>Document</title>
<style>
    [tooltip]:before {
        position: absolute;
        content: attr(tooltip);
        opacity: 1;
        color: black;
        background-color: darkgray;
        width: 10%;
    }

    [tooltip]:hover:before {
        opacity: 1;
    }
</style>
