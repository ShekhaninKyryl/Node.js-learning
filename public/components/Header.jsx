import React, {useState, useEffect} from 'react';
import Logout from './Logout.jsx';
import axios from "axios";


function Header(props) {
  let {
    login,
    logoutFn,
    getUser
  } = props;
  const [id, setId] = useState(0);
  const [name, setName] = useState('Guest');
  const [email, setEmail] = useState('');

  const IsLogout = (isLogout) => {
    let {logoutFn} = isLogout;
    if (logoutFn) {
      return (
        <Logout logoutFn={logoutFn}/>
      );
    } else {
      return null;
    }
  };

  useEffect(() => {
    axios.get('/api/user')
      .then(response => {
        console.log('Get user :', response.data);
        let {name, email, id} = response.data;
        setId(id);
        setName(name);
        setEmail(email);
        getUser(true);
      })
      .catch(error => {
        setId(0);
        setName('Guest');
        setEmail('');
        getUser(false);
        console.log('Get user error:', error.response);
      });
  }, [login]);

  return (
    <div>
      <span>
      {`Hello ${name}. ${email ? 'email: ' + email : ''}`}
      </span>
      <span >
        <IsLogout logoutFn={logoutFn}/>
      </span>
    </div>
  )

}

export default Header;
