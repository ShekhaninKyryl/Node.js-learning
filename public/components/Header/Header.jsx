import React, {useState, useEffect, useLayoutEffect} from 'react';
import Logout from '../LoginLogout/Logout.jsx';
import axios from "axios";
import {connect} from "react-redux";
import {getUserInfo} from "../../reducers/tracks/headerTracks";


function Header(props) {
    let {isLogin} = props.api;
    let {name, email, id} = props.user;


    const IsLogout = () => {
        if (isLogin) {
            return <Logout/>;
        } else {
            return null;
        }
    };
    useEffect(() => {
        props.getUser();
    }, [isLogin]);


    return (
        <div className='header-container'>
      <span className='header-span span-user-name'>
      {`Hello ${name}`}
      </span>
            <span className='header-span span-user-email'>
                {email ? `Email: ${email}` : ''}
        </span>
        <IsLogout/>
        </div>

    )

}

export default connect(
    state => ({
        api: state.api,
        user: state.user
    }),
    dispatch => ({
        getUser: () => dispatch(getUserInfo()),
    })
)(Header);
