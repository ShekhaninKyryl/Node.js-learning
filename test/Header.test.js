import React from 'react';
import {shallow} from 'enzyme';

import Header from '../public/components/Header/Header.jsx';
import Logout from '../public/components/LoginLogout/Logout.jsx';
import configureStore from 'redux-mock-store'
import {BrowserRouter as Router} from 'react-router-dom';
import {Provider} from 'react-redux';

const initialState = {
  api: {
    isLogin: false
  },
  user: {
    id: 0,
    name: 'Guest',
    email: ''
  }
};
const mockStore = configureStore();
let store;
let header;
let logout;

beforeEach(() => {
  store = mockStore(initialState);
  header = shallow(<Provider store={store}><Router><Header/></Router></Provider>);
});

it('isLogin - false', () => {
  expect(header.render()).toMatchSnapshot();
  //expect(header.render()).toEqual(true);
});

it('isLogin - true', () => {
  const initialState = {
    api: {
      isLogin: true
    },
    user: {
      id: 1,
      name: '1',
      email: '1@1.com'
    }
  };
  store = mockStore(initialState);
  header = shallow(<Provider store={store}><Router><Header/></Router></Provider>);

  expect(header.render()).toMatchSnapshot();
});
