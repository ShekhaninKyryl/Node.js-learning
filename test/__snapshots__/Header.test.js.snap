// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isLogin - false 1`] = `
<div
  class="header-container"
>
  <span
    class="header-span span-user-name"
  >
    Hello Guest
  </span>
  <span
    class="header-span span-user-email"
  />
</div>
`;

exports[`isLogin - true 1`] = `
<div
  class="header-container"
>
  <span
    class="header-span span-user-name"
  >
    Hello 1
  </span>
  <span
    class="header-span span-user-email"
  >
    Email: 1@1.com
  </span>
  <button
    class="my-button logout-button"
  >
    <a
      href="/guest"
    >
      Logout
    </a>
  </button>
</div>
`;
